backend/remote/backend.go

```diff
@@ -291,7 +291,7 @@ func (b *Remote) Configure(obj cty.Value) tfdiags.Diagnostics {
 
 	// Configure a local backend for when we need to run operations locally.
 	b.local = backendLocal.NewWithBackend(b)
-	b.forceLocal = os.Getenv("TF_FORCE_LOCAL_BACKEND") != ""
+	b.forceLocal = !entitlements.Operations || os.Getenv("TF_FORCE_LOCAL_BACKEND") != ""
```

Comment:
I thought when we discussed this before we'd said this was going to be a workspace-level "supports remote operations" flag, with this entitelement-handling logic living in the server rather than in the client. Did something change in the mean time?

Reply:
Yes and no 😏 We are still checking the workspaces `operations` attribute to determine if we should do remote operations for that specific workspace. But it turns out that is not enough.

Until yesterday my assumption was that the `operations` attribute would be a dynamic attribute whose value was a result of checking both the organization entitlements (does this org have rights to do remote operations) and a value configured on the workspace to indicate if the workspace should use remote operations or not (when remote operations are allowed).

But it turns out that this would not be straight forward to code into the backend so we ended up deciding that I should check both values in the remote backend.

This may not be the most ideal solution, but it does still take care of the main argument you shared in that discussion: `right now it would just check this organization-level entitlement on the client's behalf, but this leaves our options open for the future to e.g. allow users to turn off remote operations for particular workspaces even if they are entitled to use remote operations`.

By checking both we now support this use case, but maybe at the costs of adding a little bit of complexity. But to be honest, I don't think we can really prevent that at this point given the current boundaries within which we need to make this work.

Reply:
The main thing I worry about here is that after we ship this we are committed to always returning this entitlement from this endpoint even if in future the logic for selecting this becomes more dynamic than just checking the entitlement.

Worst case I suppose we would have the server just return this entitlement for _all_ organizations to effectively nullify this check and then make the `operations` attribute dynamic. That doesn't seem ideal to me, but I'll leave you to make that tradeoff since you have more context than I do.

I have no other feedback here, so feel free to take James's approval and move ahead with this.

Reply:
>The main thing I worry about here is that after we ship this we are committed to always returning this entitlement from this endpoint even if in future the logic for selecting this becomes more dynamic than just checking the entitlement.

I get that, but the `entitlements` endpoint is a new endpoint that was specifically introduced to support a more flexible way to manage entitlements with the new `free organizations` plan and possibly other, more flexible, plans being added to TFE next year.

Also this endpoint is now part of the V2 API, so as long as we answer to `tfe.v2` service discover requests, we can expect this API to be there.